jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/CloseStatus.java

```diff
@@ -168,21 +168,14 @@ public CloseStatus(ByteBuffer payload)
                 data.get(reasonBytes, 0, len);
 
                 // Spec Requirement : throw BadPayloadException on invalid UTF8
-                try
-                {
-                    Utf8StringBuilder utf = new Utf8StringBuilder();
-                    // if this throws, we know we have bad UTF8
-                    utf.append(reasonBytes, 0, reasonBytes.length);
-                    String reason = utf.toString();
-
-                    this.code = statusCode;
-                    this.reason = reason;
-                    return;
-                }
-                catch (Utf8Appendable.NotUtf8Exception e)
-                {
-                    throw new BadPayloadException("Invalid CLOSE Reason", e);
-                }
+                Utf8StringBuilder utf = new Utf8StringBuilder();
+                // if this throws, we know we have bad UTF8
+                utf.append(reasonBytes, 0, reasonBytes.length);
+                String reason = utf.takeString(() -> new BadPayloadException("Invalid UTF8 in CLOSE Reason", new CharacterCodingException()));
```

Comment:
Huh?  why add the CharacterCodingException at all?

Reply:
It was for websocket tests that were checking cause. Happy to change tests and not have double exception, but wanted to start with minimal test changes